### PR TITLE
Alex save v02

### PR DESCRIPTION
**Pull request by alexmaragko**
Save function created. It creates a multi-extension FITS file with the fit results. 
Added a pdf file output for the plot.

### CODE REVIEW

File: pahfit/PAHFITBase.py [hunk 1]
```diff
@@ -225,12 +226,84 @@ def plot(self, ax, x, y, model):
...
         Format TBD
         """
-        pass
+        # Instantiating lists
+        bb_temp, bb_amp = ([] for i in range(2))
+        pah_amp, pah_x_0, pah_fwhm = ([] for i in range(3))
+        line_name, line_amp, line_mean, line_std = ([] for i in range(4))
+        att = []
+        # TODO: find the variable that sets the components number (currently 55).
+        for i in range(55):
```
[the remark this answers is not in the document]
Comment (alexmaragko) [reply pahfit/PAHFITBase.py]: Yes, this sounds better. Will do.

**Comment by karllark** [comment]
Great work.

One general comment, the current save format is specific to FITS files.  I understand why as the number of parameters to be saved is different for different model components (e.g., one for temp and 4 for Gaussians).  And saving them as a multiextension FITS files works as a result.  But this does not use astropy.tables, making generating other formats non-trivial.  Not a problem, but something we should think about how to solve.  We can write custom writers for each format we want to have (e.g., current code for FITS, new code would be needed for ascii or ASDF).  Would be great if there was a generic astropy.table solution, but I don't know if there is.  Will investigate.

**Comment by karllark** [comment]
Ok, one idea for having a single table instead of multiple tables would be to have the following columns.  Then the single table could be written using the Table.write() command (`from astropy.table import Table`) and read using Table.read().  That would give us the ability to get different formats w/o much custom reading/writing routines.

Name, Amplitude, x_o, fwhm

And then only fill in the columns that apply for each component.  For example,

Name  Amplitude  x_o   fwhm
BB90    amp  
DF1      amp       x_o      fwhm
[ArII]     amp       x_o      fwhm    (requires translating std_dev to fwhm) 

What do you think?

**Comment by karllark** [comment]
BTW, thanks for correcting all the typos in the comments, etc.

**Comment by karllark** [comment]
It would be good to save the limits placed on each parameter as well.  Could be done by just adding two additional columns for each parameter (e.g., amp_min, amp_max).  With this information, getting close to have an *input* format as well!

**Comment by karllark** [comment]
Will (partially?) address issue #13.

**Comment by karllark** [comment]
And one more column per parameter - saying if it is fixed or not.   Then the blackbody temperature could be added as a parameter providing this critical information.

**Comment by alexmaragko** [comment]
No I don't mind the comments at all! In fact I do agree in everything you suggest. My only concern is if we want to add any future outputs e.g., MCMC parameters or even now if we want to get the S07_attenuation value, etc, that don't fit into any of the standard columns.

**Comment by karllark** [comment]
I agree that MCMC parameters will require something additional.  But that will specifically only be for output (not input), so I think that is ok.  Having the "standard" output also work for input will really be useful.

I don't understand what you mean by S07 attenuation value.  Isn't this already output as the tau_sil parameter?

**Comment by alexmaragko** [comment]
Yes I was referring to tau_sil parameter, and indeed it can be placed at the standard columns, with NaNs or zeroes to the non-relevant columns.

**Comment by karllark** [comment]
Great.  Using astropy.Table, you can set columns with no data as masked.  Check out 'missing values' in http://docs.astropy.org/en/stable/table/

**Comment by karllark** [comment]
I would imagine that the tau_sil parameter would be put in the amplitude column as this is the amplitude of the attenuation curve used.

**Comment by alexmaragko** [comment]
I have finished making an Astropy (Masked) Table with all columns discussed above. Do we want to save it using ascii.write for instance?

**Comment by karllark** [comment]
Why not just use table.write()?  Then it is not specific to ascii and can be fits or whatever.  The generic table.write uses the filename extension to pick the output format.

**Comment by karllark** [comment]
Yeah!

**Comment by karllark** [comment]
Is this pull request outdated given #43?  If so, please close (or tell me and I'll close).

**Pull request closed without merging**